src/wovnio/wovnphp/Url.php

```diff
@@ -224,6 +247,26 @@ public static function removeLangCode($uri, $lang_code, $settings)
                 // limit to one replacement
                 $prefix = empty($site_prefix_path) ? '' : '/' . $site_prefix_path;
                 return preg_replace("@$prefix/$lang_code(/|$)@i", "$prefix/", $uri, 1);
+            case 'custom_domain':
+                $customDomainLangs = $store->getCustomDomainLangs();
+                $default_lang = $settings['default_lang'];
+                $customDomainLangToRemove = $customDomainLangs->getCustomDomainLangByLang($lang_code);
+                $defaultCustomDomainLang = $customDomainLangs->getCustomDomainLangByLang($default_lang);
+                $newUri = $uri;
+                if (self::isAbsoluteUri($uri)) {
+                    $newUri = CustomDomainLangUrlHandler::changeToNewCustomDomainLang($uri, $customDomainLangToRemove, $defaultCustomDomainLang);
+                } elseif (self::isAbsolutePath($uri)) {
+                    $absoluteUrl = $headers->protocol . '://' . $headers->unmaskedHost . $uri;
+                    $absoluteUrlWithLang = CustomDomainLangUrlHandler::changeToNewCustomDomainLang($absoluteUrl, $customDomainLangToRemove, $defaultCustomDomainLang);
+                    $segments = self::makeSegmentsFromAbsoluteUrl($absoluteUrlWithLang);
+                    $newUri = $segments['others'];
+                } elseif ($uri === $customDomainLangToRemove->getHost()) {
+                    $absoluteUrl = $headers->protocol . '://' . $uri . $headers->unmaskedPathname;
```

Comment:
We need `originalHost` and `originalPath` for these cases. The $pathname, $host variables in Headers will have the language removed already, but to remove the language from `french.testsite.com/fr/`, we need to combine `french.testsite.com` with `/fr/`, not `/`

Reply:
actually, looks like $unmaskedPath and $unmaskedHost do serve this purpos. Once you merge with master you may need to re-add them. Rename them to `originalHost` etc though.